### PR TITLE
Dynamic dashboard: Support fetching variation reports

### PR DESCRIPTION
**Pull request by itsmeichigo**
Part of #12796

## Description

This PR integrates the endpoint `wc-analytics/reports/variations` to support displaying details for product variations on the stock card. Changes include:
- Updated `ProductsRemote` with a new method to load variation reports. The implementation is similar to loading product reports as the response's structure is the same, the only difference is the required param `variations` for the variation IDs to load, and the response includes variation IDs.
- Updated `ProductAction` and `ProductStore` to support loading variation reports.
- Updated unit tests.

## Testing instructions

Just CI passing is sufficient.

## Screenshots

N/A

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: Networking/Networking/Remote/ProductsRemote.swift [hunk 1]
```diff
@@ -478,6 +489,40 @@ public final class ProductsRemote: Remote, ProductsRemoteProtocol {
...
+                                     orderBy: ProductsRemote.OrderKey,
+                                     order: ProductsRemote.Order) async throws -> [ProductReport] {
+        let dateFormatter = DateFormatter.Defaults.iso8601WithoutTimeZone
+        dateFormatter.timeZone = timeZone
+        let path = Path.variationReports
+        let parameters: [String: Any] = [
+            ParameterKey.products: productIDs,
+            ParameterKey.variations: variationIDs,
+            ParameterKey.before: dateFormatter.string(from: earliestDateToInclude),
+            ParameterKey.after: dateFormatter.string(from: latestDateToInclude),
```
Comment (selanthiraiyan) [inline Networking/Networking/Remote/ProductsRemote.swift]: nit: I think `earliestDateToInclude` should be sent as `after` and `latestDateToInclude` as `before`. I left a similar comment [here](https://github.com/woocommerce/woocommerce-ios/pull/12803#discussion_r1615438165)

File: Yosemite/Yosemite/Actions/ProductAction.swift [hunk 2]
```diff
@@ -229,4 +229,28 @@ public enum ProductAction: Action {
...
+    /// - Parameter latestDateToInclude: The latest date for the product reports.
+    /// - Parameter pageNumber: The index of the content page to query.
+    /// - Parameter pageSize: The max amount of items to return per page.
+    /// - Parameter orderBy: The key to sort returned items.
+    /// - Parameter order: Whether to sort returned items ASC or DESC.
+    /// - Parameter completion: Closure to trigger when fetching completes.
+    ///
+    case fetchVariationReports(siteID: Int64,
+                               productIDs: [Int64],
+                               variationIDs: [Int64],
```
Comment (selanthiraiyan) [inline Yosemite/Yosemite/Actions/ProductAction.swift:247]: [Here](https://github.com/woocommerce/woocommerce-ios/pull/12803#discussion_r1615451213), I left a question about how the endpoint differentiates different variations when there is a conflict in variation ID value between different products. Please correct me if I am wrong.

**Review by selanthiraiyan — APPROVED** [review]
:shipit:

File: Networking/Networking/Remote/ProductsRemote.swift (hunk 1 above)
Comment (itsmeichigo) [reply Networking/Networking/Remote/ProductsRemote.swift]: Great catch! Updated for both the product and variation report requests in 11cc064dc5f98ae3bd379966794a424d6c0f654b.

File: Yosemite/Yosemite/Actions/ProductAction.swift (hunk 2 above)
Comment (itsmeichigo) [reply Yosemite/Yosemite/Actions/ProductAction.swift:247]: From the discussions in p1716794405419699-slack-C03L1NF1EA3, I believe the variation IDs are unique so we can group them in a single request.

**Pull request merged**